Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -340,4 +340,26 @@ bool Element::is_focused() const
     return document().focused_element() == this;
 }
 
+NonnullRefPtrVector<Element> Element::get_elements_by_tag_name(const FlyString& tag_name) const
+{
+    NonnullRefPtrVector<Element> elements;
+    for_each_in_subtree_of_type<Element>([&](auto& element) {
+        if (element.local_name() == tag_name)
+            elements.append(element);
+        return IterationDecision::Continue;
+    });
+    return elements;
+}
+
+NonnullRefPtrVector<Element> Element::get_elements_by_class_name(const FlyString& class_name) const
+{
+    NonnullRefPtrVector<Element> elements;
+    for_each_in_subtree_of_type<Element>([&](auto& element) {
+        if (element.has_class(class_name))
```

Comment:
In quirks modes, this should be case insensitive. From the spec:
```
The comparisons for the classes must be done in an ASCII case-insensitive manner if root’s node document’s mode is "quirks", and in an identical to manner otherwise. 
```
`root` being `this` in this case.